drivers/media/platform/raspberrypi/pisp_be/pisp_be.c

```diff
@@ -47,9 +47,9 @@ MODULE_LICENSE("GPL v2");
 #define PISP_BE_INTERRUPT_STATUS_OFFSET		0x1c
 #define PISP_BE_AXI_OFFSET			0x20
 #define PISP_BE_CONFIG_BASE_OFFSET		0x40
-#define PISP_BE_IO_INPUT_ADDR0_LO_OFFSET 	PISP_BE_CONFIG_BASE_OFFSET
-#define PISP_BE_GLOBAL_BAYER_ENABLE_OFFSET (PISP_BE_CONFIG_BASE_OFFSET + 0x70)
-#define PISP_BE_GLOBAL_RGB_ENABLE_OFFSET   (PISP_BE_CONFIG_BASE_OFFSET + 0x74)
+#define PISP_BE_IO_INPUT_ADDR0(n)		(0x40 + 8 * (n))
+#define PISP_BE_GLOBAL_BAYER_ENABLE		0xb0
+#define PISP_BE_GLOBAL_RGB_ENABLE		0xb4
```

Comment:
Should we not be using ``PISP_BE_CONFIG_BASE_OFFSET`` in these macros, e.g.:
```
#define PISP_BE_IO_INPUT_ADDR0(n) (PISP_BE_CONFIG_BASE_OFFSET+ 8 * (n))
```

Reply:
It's a style question -- the original was derived from our hardware specification and test code, which uses offsets wrt the block base address, but wasn't particularly Linux-style.

Since we ignore the user-specified configuration here and write directly to the registers, either style of basing seems acceptable to me.

I agree that a parameterized macro would be nice, though FWIW I might remove the `0` (the parameter _n_ replaces the 0) and add a `LO`/`HI` distinction for the two 32-bit halves.

Reply:
I kept `INPUT` as it's actually the register name used in the specs, I could drop it but matching specs seems a good idea. I agree we can create a `LO/HI` version though.

To be honest I would remove `PISP_BE_CONFIG_BASE_OFFSET` completely but replacing 

```
	begin =	offsetof(struct pisp_be_config, global.bayer_order)
	      / sizeof(u32);
	end = offsetof(struct pisp_be_config, axi) / sizeof(u32);
	for (unsigned int u = begin; u < end; u++)
		pispbe_wr(pispbe, PISP_BE_CONFIG_BASE_REG + 4 * u,
			  ((u32 *)job->config)[u]);
```

seems not a good idea, but I would only use `PISP_BE_CONFIG_BASE_OFFSET` here and not for defining other registers where we write directly to.

Reply:
I was being a bit pedantic: `INPUT0_LO ` is the name of the _first_ of a series of address registers, it's not the name of _all_ of the address registers. Maybe suitable macro names might be something like `PISP_BE_DMA_ADDRESS_LO(n)` and `PISP_BE_DMA_ADDRESS_HI(n)`?

BTW I think we made a mistake when we included DMA addresses in `struct pisp_be_config` -- they are not for the user to specify -- but now that it's in UAPI, we can't change the structure! We just have to skip over them. I'm not sure how to make it cleaner.

Reply:
> I was being a bit pedantic: `INPUT0_LO ` is the name of the _first_ of a series of address registers, it's not the name of _all_ of the address registers. Maybe suitable macro names might be something like `PISP_BE_DMA_ADDRESS_LO(n)` and `PISP_BE_DMA_ADDRESS_HI(n)`?

Yeah, I realized it later, so I dropped `INPUT` and `0` from name. I can rename it it `PISP_BE_DMA_ADDRESS...()` if preferred

> 
> BTW I think we made a mistake when we included DMA addresses in `struct pisp_be_config` -- they are not for the user to specify -- but now that it's in UAPI, we can't change the structure! We just have to skip over them. I'm not sure how to make it cleaner.

Well, the uAPI is not upstream yet, so we're still in time to change it :)